File: app/src/main/java/se/jakob/knarkklocka/data/AlarmHistoryViewModelFactory.java
package se.jakob.knarkklocka.data;

import android.arch.lifecycle.ViewModelProvider;

public class AlarmHistoryViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private final AlarmDatabase mDb;
    private final int mTaskId;


    public AlarmHistoryViewModelFactory(AlarmDatabase database, int taskId) {
        mDb = database;
        mTaskId = taskId;
    }
/*
    @NonNull
    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        //noinspection unchecked
        return (T) new AlarmHistoryViewModel(mDb, mTaskId);
    }*/
}

File: app/src/main/java/se/jakob/knarkklocka/viewmodels/AlarmHistoryViewModelFactory.kt
package se.jakob.knarkklocka.viewmodels

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

import se.jakob.knarkklocka.data.AlarmRepository

/**
 * Factory for creating a [AlarmHistoryViewModel] with a constructor that takes a
 * [AlarmRepository].
 */

class AlarmHistoryViewModelFactory(
        private val alarmRepository: AlarmRepository
) : ViewModelProvider.NewInstanceFactory() {

    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return AlarmHistoryViewModel(alarmRepository) as T
    }
}
